src/Workspaces/CSharpTest/EmbeddedLanguages/RegularExpressions/CSharpRegexParserTests_BasicTests.cs

```diff
@@ -0,0 +1,15106 @@
+  </CompilationUnit>
+  <Diagnostics>
+    <Diagnostic Message=""Unterminated (?#...) comment"" Start=""9"" Length=""4"" />
+  </Diagnostics>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..13)"" />
+  </Captures>
+</Tree>", RegexOptions.None);
+        }
+
+        [Fact]
+        public void TestMultipleComments1()
+        {
+            Test(@"""(?#)(?#)""", @"<Tree>
+  <CompilationUnit>
+    <Sequence />
+    <EndOfFile>
+      <Trivia>
+        <CommentTrivia>(?#)</CommentTrivia>
+        <CommentTrivia>(?#)</CommentTrivia>
+      </Trivia>
+    </EndOfFile>
+  </CompilationUnit>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..17)"" />
+  </Captures>
+</Tree>", RegexOptions.None);
+        }
+
+        [Fact]
+        public void TestMultipleComments2()
+        {
+            Test(@"""(?#)(?#)""", @"<Tree>
+  <CompilationUnit>
+    <Sequence />
+    <EndOfFile>
+      <Trivia>
+        <CommentTrivia>(?#)</CommentTrivia>
+        <CommentTrivia>(?#)</CommentTrivia>
+      </Trivia>
+    </EndOfFile>
+  </CompilationUnit>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..17)"" />
+  </Captures>
+</Tree>", RegexOptions.IgnorePatternWhitespace);
+        }
+
+        [Fact]
+        public void TestMultipleComments3()
+        {
+            Test(@"""(?#) (?#)""", @"<Tree>
+  <CompilationUnit>
+    <Sequence>
+      <Text>
+        <TextToken>
+          <Trivia>
+            <CommentTrivia>(?#)</CommentTrivia>
+          </Trivia> </TextToken>
+      </Text>
+    </Sequence>
+    <EndOfFile>
+      <Trivia>
+        <CommentTrivia>(?#)</CommentTrivia>
+      </Trivia>
+    </EndOfFile>
+  </CompilationUnit>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..18)"" />
+  </Captures>
+</Tree>", RegexOptions.None);
+        }
+
+        [Fact]
+        public void TestMultipleComments4()
+        {
+            Test(@"""(?#) (?#)""", @"<Tree>
+  <CompilationUnit>
+    <Sequence />
+    <EndOfFile>
+      <Trivia>
+        <CommentTrivia>(?#)</CommentTrivia>
+        <WhitespaceTrivia> </WhitespaceTrivia>
+        <CommentTrivia>(?#)</CommentTrivia>
+      </Trivia>
+    </EndOfFile>
+  </CompilationUnit>
+  <Captures>
+    <Capture Name=""0"" Span=""[9..18)"" />
+  </Captures>
+</Tree>", RegexOptions.IgnorePatternWhitespace);
+        }
+
+        [Fact]
+        public void TestCommentAfterEscapeInCharacterClass1()
+        {
+            Test(@"@""[a\p{Lu}(?#)b]""", @"<Tree>
+  <CompilationUnit>
+    <Sequence>
+      <CharacterClass>
+        <OpenBracketToken>[</OpenBracketToken>
+        <Sequence>
+          <Text>
+            <TextToken>a</TextToken>
+          </Text>
+          <CategoryEscape>
+            <BackslashToken>\</BackslashToken>
+            <TextToken>p</TextToken>
+            <OpenBraceToken>{</OpenBraceToken>
+            <EscapeCategoryToken>Lu</EscapeCategoryToken>
+            <CloseBraceToken>}</CloseBraceToken>
+          </CategoryEscape>
+          <Text>
+            <TextToken>(?#)b</TextToken>
```

Comment:
Should this be trivia (comment)?

Reply:
Nope.  Because it's in a char-class.  And that's what thsi test is validating, that it isn't treated as a comment in that case.

Reply:
Updated test names to make it clearer this is intentional.